src/resolve/resolved-dns-packet.c

```diff
@@ -47,13 +50,14 @@ int dns_packet_new(DnsPacket **ret, DnsProtocol protocol, size_t mtu) {
 
         assert(ret);
 
-        if (mtu <= UDP_PACKET_HEADER_SIZE)
+        /* When dns_packet_new() is called with mtu == 0, allocate more than the
+         * absolute minimum (which is the dns packet header size), to avoid
+         * resizing immediately again after appending the first data to the packet.
+         */
+        if (mtu < UDP_PACKET_HEADER_SIZE)
```

Comment:
@poettering If argument `mtu` pass to the `dns_packet_new()` is equal to 50 then the data that need to be allocated is only 22, since the UDP header size is equal to 28 (if I am not wrong, this is not really important, this is just for the example)

So when the parameter `mtu` is passed to  `dns_packet_new()` with value 50, what is the expected allocated size ?